lib/pbench/test/unit/server/query_apis/test_datasets_delete.py

```diff
@@ -188,8 +188,34 @@ def test_no_dataset(
         assert response.status_code == HTTPStatus.NOT_FOUND
         assert response.json["message"] == "Dataset 'badwolf' not found"
 
+    def test_no_index(
+        self, client, monkeypatch, attach_dataset, pbench_token, server_config
+    ):
+        """
+        Check the delete API if the dataset has no INDEX_MAP. It should
+        succeed without tripping over Elasticsearch.
```

Comment:
I don't see any positive assertion that we didn't call Elasticsearch, and I don't see any provision for detecting if a request to it were made...so how do we know that this test is driving the desired behavior?  (I mean, yeah, we prove that the dataset was successfully deleted, which is important, but I don't see how we're ensuring that there is no `INDEX_MAP` and I don't see how we know whether it called ES or not...what am I missing?  I note that this test does _not_ include the `get_document_map` fixture, but I don't understand the implication of running without it...is that what ensures the lack of a map??  If so, it would be helpful to include that detail in the docstring....)

Reply:
Without an index map, it would have blown up trying to iterate through the map. I decided not to add a lot of extra baggage.